### DISCUSSION TITLE
About addGeneIntegrationMatrix

### DISCUSSION THREAD

**Discussion by tzhu-bio** [GreenleafLab/ArchR#1023] [Questions / Documentation]
When I used **addGeneIntegrationMatrix** to integrate the ATAC and RNA, it only found 15605 overlapping gene names from gene scores and rna matrix! Is this a relatively few overlapping genes?

And I found the GeneIntegrationMatrix features only has the "ENSMFAGxxxxxxxx" gene id but with no gene name like "PGBD2"....

![image](https://user-images.githubusercontent.com/54537420/132526728-026a63ae-a8c3-40f4-a349-cebbb0ae559f.png) 

Then I change the rna matrix gene name, which are the same as genescore matrix gene name, but the result did not change.

**Comment by rcorces** [comment]
This behavior is not observed using the tutorial data on my end. `getFeatures()` returns gene symbols for me. What is your reference genome? The names of the features in GeneIntegrationMatrix would be based off gene names that were set using your reference genome.

```
> getFeatures(ArchRProj = projHeme5, useMatrix = "GeneIntegrationMatrix")
   [1] "OR4F5"        "LINC00115"    "FAM41C"       "SAMD11"       "NOC2L"        "KLHL17"       "PLEKHN1"     
   [8] "HES4"         "ISG15"        "AGRN"         "RNF223"       "C1orf159"     "TTLL10"       "TNFRSF18"    
  [15] "TNFRSF4"      "SDF4"         "B3GALT6"      "UBE2J2"       "SCNN1D"       "ACAP3"        "PUSL1"     
...
```

**Reply by tzhu-bio** [reply]
I created a custom ArchRGenome, the result of getFeatures in my custom ArchRGenome with no problem. But in my RNA matrix, gene names are "ENSMFAGxxxxxxxx" format. I transformed the gene name to match with ATAC gene names. But I got an error in **addGeneIntegrationMatrix**.

![image](https://user-images.githubusercontent.com/54537420/132856737-b12ebeff-a800-4dd5-a66f-b4afc7b16157.png)

[ArchR-addGeneIntegrationMatrix-3fb017c6130e0-Date-2021-09-08_Time-21-29-51.log](https://github.com/GreenleafLab/ArchR/files/7143914/ArchR-addGeneIntegrationMatrix-3fb017c6130e0-Date-2021-09-08_Time-21-29-51.log)

**Reply by rcorces** [reply]
what about the names of the features in the RNA object you are trying to integrate with? The "ENSMFAGxxxxxxxx" names must be coming from somewhere.

**Reply by tzhu-bio** [reply]
The names of the features in RNA object are from gff (https://m.ensembl.org/Macaca_fascicularis/Info/Annotation) and they are "ENSMFAGxxxxxxxx" format. But in ATAC, I created custom genome annotation with the following dataframe. (If a "ENSMFAG0000001" has a gene name like "PGBD2", I used "PGBD2" instead of "ENSMFAG0000001") 
![image](https://user-images.githubusercontent.com/54537420/132933804-e8102ae2-61c7-4a5b-95e0-ce2172eceeec.png)

So I need to change all the "ENSMFAGxxxxxxx"  to gene names if they have gene names like PGBD2 in RNA object before I run addGeneIntegrationMatrix. But I got an error (shown above) after I change the gene format.

**Reply by rcorces** [reply]
I dont think I understand your questions. There is clearly some incompatibility between the gene names in your custom genome annotation and the gene names in your RNA object. We dont have any experience ourselves with custom genomes and, as I'm sure you can imagine, they each have their own quirks. My recommendation is that you look at the tutorial data provided and deduce what your genome annotation and RNA objects should look like.

**Reply by tzhu-bio** [reply]
Thanks for your reply.

**Comment by tzhu-bio** [comment]
Sorry to bother you again.

 I got the features from the GeneScore matrix and why they all start with "chr".

![image](https://user-images.githubusercontent.com/54537420/133016852-5871a584-5806-4086-be88-a091a9ac8e90.png)

This is not the same as the tutorial. I don't think I made a mistake in constructing the custom genome.

**Reply by rcorces** [reply]
Sorry - I'm not sure. What does your custom genome annotation object look like? Run:
`getGeneAnnotation(ArchRProj)$genes`

**Reply by tzhu-bio** [reply]
![image](https://user-images.githubusercontent.com/54537420/133172608-ccedb9eb-450a-4db1-b66b-04e6f34487d8.png)

This looks normal.

**Reply by rcorces** [reply]
what do you see when you do:
`ArchR:::.getFeatureDF(getArrowFiles(mcl_motif), "GeneScoreMatrix")`

**Reply by tzhu-bio** [reply]
![image](https://user-images.githubusercontent.com/54537420/133276428-4903e18f-cd44-4510-bc6a-24914fab287f.png)

**Reply by tzhu-bio** [reply]
I noticed that some chromosome names do not start with "chr" when I build the genome, is that the reason?

![image](https://user-images.githubusercontent.com/54537420/133276975-814e53e7-8616-48f1-9809-ecb4274afbcd.png)

![image](https://user-images.githubusercontent.com/54537420/133277130-0c8433df-2406-469d-af9d-86341eaeb613.png)

**Reply by rcorces** [reply]
I think I'm not really following your issue very well. From what I gather re-reading this thread:

1) Your GeneScoreMatrix has genes named with both "ENSMFA####" and regular gene names like "PGBD2"
2) Your RNA object only has gene names like "ENSMFA####"
3) When you overlap these, you only get ENSMFA#### genes

This should not be surprising - those are the only gene names that overlap. You need to make sure that whatever gene annotation you use in your custom gene annotation matches what you use in your RNA.
